### PR TITLE
Add func wrapper for converting shapes to ivy shapes

### PR DESCRIPTION
**Pull request by vaithak**

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by Sam-Armstrong — COMMENTED** [review]
lgtm, but maybe @AnnaTz can quickly check this as I'm not very familiar with this part of the codebase?

File: ivy/functional/frontends/torch/func_wrapper.py [hunk 1]
```diff
@@ -243,6 +243,40 @@ def outputs_to_native_arrays_torch(*args, **kwargs):
     return outputs_to_native_arrays_torch
 
 
+def to_ivy_shape(fn: Callable) -> Callable:
+    """
+    Wrap `fn` so it receives `ivy.Shape` instead of Native shapes.
+
+    Wrap `fn` so that any `torch.Size` arguments are converted to
+    `ivy.Shape` instances.
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/func_wrapper.py:251]: To be clear, please refer to the shapes we are converting as `torch_frontend.Size`, not `torch.Size` (this refers to the original torch class) neither NativeShape (this refers mostly to the backends).
Also the first sentence is not necessary at all.

File: ivy/functional/frontends/torch/func_wrapper.py [hunk 2]
```diff
@@ -243,6 +243,40 @@ def outputs_to_native_arrays_torch(*args, **kwargs):
...
+        new_kwargs = {
+            key: (
+                ivy.to_ivy_shape(tuple(value))
+                if key in ["shape", "size"]
+                and isinstance(value, ivy.functional.frontends.torch.Size)
+                else value
+            )
+            for key, value in kwargs.items()
+        }
+        # if any of the args are instance of ivy.NativeShape, convert them to ivy.Shape.
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/func_wrapper.py:265]: Same here. Please change to `torch_frontend.Size`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by AnnaTz** [comment]
Tagging @vaithak to check the transpilation cases mentioned in the trello card.

**Comment by vaithak** [comment]
> Tagging @vaithak to check the transpilation cases mentioned in the trello card.

Verified the transpilation of Bart and Levit models 👍🏼

**Review by AnnaTz — APPROVED** [review]

**Pull request merged**